src/Stomp/Network/Connection.php

```diff
@@ -125,7 +125,7 @@ public function __construct($brokerUri, $connectionTimeout = 1, array $context =
         $this->parser = new Parser();
         $this->connectTimeout = $connectionTimeout;
         $this->context = $context;
-        $pattern = "|^(([a-zA-Z0-9]+)://)+\(*([a-zA-Z0-9\.:/i,-]+)\)*\??([a-zA-Z0-9=&]*)$|i";
+        $pattern = "|^(([a-zA-Z0-9]+)://)+\(*([a-zA-Z0-9\.:/i,-_]+)\)*\??([a-zA-Z0-9=&_]*)$|i";
```

Comment:
you also added the `_` as allowed character within connection properties... was it intended?